### PR TITLE
[Model] Support Qwen2ForSequenceClassification

### PR DESCRIPTION
**Pull request by Ximingwang-09**
## Motivation

Implement the Qwen2ForSequenceClassification.
## Modifications

Key changes:
1. Implement Qwen2ForSequenceClassification model architecture
2. Add model registration for sequence classification variant
3. Update documentation to reflect new sequence classification capability
4. Usecase is similar like https://github.com/sgl-project/sglang/pull/3772

```
python -m sglang.launch_server \
	--model jason9693/Qwen2.5-1.5B-apeach \
	--trust-remote-code \
        --is-embedding \
	--served-model-name Qwen2.5-1.5B-APEACH \
	--port 8080
```
Test Result
```
Loading checkpoint shards: 100%|██████████████████████████████████████████████| 14/14 [00:17<00:00,  1.28s/it]
Loading safetensors checkpoint shards:   0% Completed | 0/14 [00:00<?, ?it/s]
Loading safetensors checkpoint shards: 100% Completed | 14/14 [00:00<00:00, 263.87it/s]

hf_scores=tensor([-1.7803, -2.4863])
srt_scores=tensor([-1.7520, -2.4961])
.
----------------------------------------------------------------------
Ran 1 test in 74.832s

OK
```
## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

**Review by zhaochenyang20 — CHANGES_REQUESTED** [review]
Could you change the unit test in `test/srt/models/test_embedding_models.py`?

**Review by zhaochenyang20 — APPROVED** [review]
LGTM, will merge after the CI

File: test/srt/models/test_embedding_models.py [hunk 1]
```diff
@@ -91,7 +93,12 @@ def assert_close_prefill_logits(
                 ), "embeddings are not all close"
 
     def test_prefill_logits(self):
-        for model, tp_size, prefill_tolerance in MODELS:
+        models_to_test = MODELS
+
+        if is_in_ci():
+            models_to_test = [random.choice(MODELS)]
```
Comment (zhyncs) [inline test/srt/models/test_embedding_models.py:99]: Why use `random.choice` here?

Comment (Ximingwang-09) [reply test/srt/models/test_embedding_models.py:99]: @zhaochenyang20  suggested streamlining CI workflows to reduce system pressure.

Comment (zhaochenyang20) [reply test/srt/models/test_embedding_models.py:99]: @zhyncs I ask him to add this since we have to many models to check in this test. each model take 40s and we can not always increase the model. so we randomly choose one model to test in CI and locally we test them all. we do this for verl-engine also. If we run each CI 100 times per day, we can still catch the error. do you agree with this?

**Comment by zhaochenyang20** [comment]
@Ximingwang-09 great! we merged. love to see the collaboration with ant group and sglang!

**Comment by Ximingwang-09** [comment]
> @Ximingwang-09 great! we merged. love to see the collaboration with ant group and sglang!

Appreciate it! SGLang is an excellent open-source project. Excited for more synergies ahead!

File: test/srt/models/test_embedding_models.py (hunk 1 above)
Comment (yiakwy-xpu-ml-framework-team) [reply test/srt/models/test_embedding_models.py:99]: Yes I agree regression tests of models are very expensive. Previously we will program manager to track the model performance from different teams and trigger it before half month release (yes disable it in nightly build). 

Randomness is a good way to cover alll topics without running all regress tests for each PR. However it reduces the posibility to produce a problem. 

Engineering is all about comproise, how about this plan:

- **nightly build** : only cirticle models , let's limited to top K (3 or 5 ? ) MoE (DeepSeek, QwenMax, Qwen light), Llama Dense, VLM (pending to be supported)

- **bi-weekly release** regression tests (it seems frequency how we release major updates) : reseve all machines for weekends, and then trigger all heavy regression tests items to focus (because regression is really expensive).

In Phabricator, or KanBan system, we typically maintain a PRs (diffs) relationship map, so that we can do some optimization among PRs to reduce total engineering efforts when merging something. 

With this graph, we can also be more confident in coordinating differnt PRs, tasks if CI occasionally failed (it did happened), so everyone will not be left behind work towards a common goal D

**Comment by Alcanderian** [comment]
Hi, @Ximingwang-09 @zhyncs , the ci test has failed and I comfrim it by modifying `test/srt/models/test_embedding_models.py:TestEmbeddingModels::test_prefill_logits` to:
```
    def test_prefill_logits(self):
        models_to_test = MODELS

        if is_in_ci():
            models_to_test = [random.choice(MODELS)]

        models_to_test = [MODELS[-1]]

        for model, tp_size, prefill_tolerance in models_to_test:
            for torch_dtype in TORCH_DTYPES:
                self.assert_close_prefill_logits(
                    DEFAULT_PROMPTS, model, tp_size, torch_dtype, prefill_tolerance
                )
```

refer to: https://github.com/sgl-project/sglang/actions/runs/14264664040/job/39983665710?pr=5059

We may have to fix it LOL

**Comment by Ximingwang-09** [comment]
> Hi, @Ximingwang-09 @zhyncs , the ci test has failed and I comfrim it by modifying `test/srt/models/test_embedding_models.py:TestEmbeddingModels::test_prefill_logits` to:
> 
> ```
>     def test_prefill_logits(self):
>         models_to_test = MODELS
> 
>         if is_in_ci():
>             models_to_test = [random.choice(MODELS)]
> 
>         models_to_test = [MODELS[-1]]
> 
>         for model, tp_size, prefill_tolerance in models_to_test:
>             for torch_dtype in TORCH_DTYPES:
>                 self.assert_close_prefill_logits(
>                     DEFAULT_PROMPTS, model, tp_size, torch_dtype, prefill_tolerance
>                 )
> ```
> 
> refer to: https://github.com/sgl-project/sglang/actions/runs/14264664040/job/39983665710?pr=5059
> 
> We may have to fix it LOL
Yes, I think we should remove the `jason9693/Qwen2.5-1.5B-apeach` from the embedding model test, as it should be in the reward model test.

**Comment by Ximingwang-09** [comment]
> Hi, @Ximingwang-09 @zhyncs , the ci test has failed and I comfrim it by modifying `test/srt/models/test_embedding_models.py:TestEmbeddingModels::test_prefill_logits` to:
> 
> ```
>     def test_prefill_logits(self):
>         models_to_test = MODELS
> 
>         if is_in_ci():
>             models_to_test = [random.choice(MODELS)]
> 
>         models_to_test = [MODELS[-1]]
> 
>         for model, tp_size, prefill_tolerance in models_to_test:
>             for torch_dtype in TORCH_DTYPES:
>                 self.assert_close_prefill_logits(
>                     DEFAULT_PROMPTS, model, tp_size, torch_dtype, prefill_tolerance
>                 )
> ```
> 
> refer to: https://github.com/sgl-project/sglang/actions/runs/14264664040/job/39983665710?pr=5059
> 
> We may have to fix it LOL

Yes, I think we should remove the `jason9693/Qwen2.5-1.5B-apeach` from the embedding model test, as it should be in the reward model test.
